types/database/channel.d.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Query, Relation} from '@nozbe/watermelondb';
+import Model, {Associations} from '@nozbe/watermelondb/Model';
+
+import ChannelInfo from '@typings/database/channel_info';
+import ChannelMembership from '@typings/database/channel_membership';
+import Draft from '@typings/database/draft';
+import GroupsInChannel from '@typings/database/groups_in_channel';
+import MyChannel from '@typings/database/my_channel';
+import MyChannelSettings from '@typings/database/my_channel_settings';
+import Post from '@typings/database/post';
+import PostsInChannel from '@typings/database/posts_in_channel';
+import Team from '@typings/database/team';
+import User from '@typings/database/user';
+
+/**
+ * The Channel model represents a channel in the Mattermost app.
+ */
+export default class Channel extends Model {
+    /** table (entity name) : Channel */
+    static table: string;
+
+    /** associations : Describes every relationship to this entity. */
+    static associations: Associations;
+
+    /** create_at : The creation date for this channel */
+    createAt: number;
+
+    /** creator_id : The user who created this channel */
+    creatorId: string;
+
+    /** delete_at : The deletion/archived date of this channel */
+    deleteAt: number;
+
+    /** display_name : The channel display name (e.g. Town Square ) */
+    displayName: string;
+
+    /** is_group_constrained : If group is restricted to certain users/teams only */
+    isGroupConstrained: boolean;
+
+    /** name : The name of the channel (e.g town-square) */
+    name: string;
+
+    /** team_id : The team to which this channel belongs.  It can be empty for direct/group message. */
+    teamId: string;
+
+    /** type : The type of the channel ( e.g. G: group messages, D: direct messages, P: private channel and O: public channel) */
+    type: string;
+
+    /** members : Users belonging to this channel */
+    members: ChannelMembership[];
+
+    /** draft : All drafts for this channel */
+    drafts: Draft[];
+
+    /** groupsInChannel : Every group contained in this channel */
```

Comment:
@hahmadia is this correct?

Reply:
Do we know what endpoint is going to populate this? I couldn't find this key in our backend so I'm not 100% where the data populating this would be coming from. Once I know it, I can give a better answer.

Reply:
Please do tag when you respond so I can get GitHub notification

Reply:
@hahmadia not sure about the endpoint but as far as I'm aware we have group constraints for specific channels as well as teams.

This was based on the redux store

Reply:
Could you point me to the part of the redux store this was based off of? Gut feeling is saying this has something to do with group mentions

Reply:
@hahmadia I think your gut feeling my be wrong? app/mm-redux/reducers/entities/groups.ts

Reply:
Ok your good then.